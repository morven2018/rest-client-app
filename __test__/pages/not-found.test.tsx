import { render, screen } from '@testing-library/react';
import NotFound from '@/app/not-found';

jest.mock('next/link', () => {
  const MockLink = ({
    href,
    children,
  }: {
    href: string;
    children: React.ReactNode;
  }) => <a href={href}>{children}</a>;
  MockLink.displayName = 'MockLink';
  return MockLink;
});

describe('NotFound Page', () => {
  it('should render without throwing errors', () => {
    expect(() => render(<NotFound />)).not.toThrow();
  });

  it('should display the error message', () => {
    render(<NotFound />);

    expect(screen.getByText('Oooops!')).toBeTruthy();
    expect(screen.getByText('This page is not found')).toBeTruthy();
  });

  render(<NotFound />);

  const homeLink = screen.getByText('Home');
  expect(homeLink).toBeTruthy();
});

describe('NotFound Page Snapshot', () => {
  it('should match snapshot', () => {
    const { container } = render(<NotFound />);
    expect(container).toMatchSnapshot();
  });
});
