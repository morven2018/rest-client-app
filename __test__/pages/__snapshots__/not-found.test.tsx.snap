// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NotFound Page Snapshot should match snapshot 1`] = `
<div>
  <main
    class="flex flex-col items-center justify-center min-h-screen gap-5"
  >
    <h2
      class="text-red-600 font-bold text-5xl"
    >
      Oooops!
    </h2>
    <h2
      class="text-xl"
    >
      This page is not found
    </h2>
    <a
      href="/"
    >
      Home
    </a>
  </main>
</div>
`;
